sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -193,6 +193,15 @@ public ComponentDto selectOrFailByKey(DbSession session, String key) {
     return Optional.fromNullable(mapper(session).selectByKey(key));
   }
 
+  /**
+   * Select all Components in DB with the specified key.
+   *
+   * Used by Developer Cockpit
+   */
+  public List<ComponentDto> selectByKeyWithDuplicates(DbSession session, String key) {
```

Comment:
I'll use the existing method, checking ITs right now on Dev Cockpit
I'll close this PR then